common/views.py

```diff
@@ -129,3 +133,80 @@ def post(self, request, *args, **kwargs):
+    
+    profile_edit_form_class = ProfileEditForm
+    user_edit_form_class= UserEditForm
+    
+    def get_user_profile_info(self, request):
+         # Get the user profile
+        try:
+        # Note that not all users have a user profile object
+        # such as the admin superuser
+            user_profile_data = UserProfile.objects.get(user_id=request.user.id)
+        except ObjectDoesNotExist as e:
+            logger.error(str(e), exc_info=True)
+            user_profile = UserProfile()
+        except Exception as e:
+            logger.error(str(e), exc_info=True)
+            user_profile = UserProfile()
+            
+        return user_profile_data
+        
+    
+    def get(self, request, *args, **kwargs):
+        
+        user_profile_data = self.get_user_profile_info(request)
+        
+        profile_edit_form = self.profile_edit_form_class(initial={
+                                                    "affiliation" : user_profile_data.affiliation,
+                                                    "department" : user_profile_data.department
+                                                 }) 
+        user_edit_form = self.user_edit_form_class(initial={
+                                                    "email" : user_profile_data.user.email,
+                                                    "first_name" : user_profile_data.user.first_name,
+                                                    "last_name" : user_profile_data.user.last_name
+                                                 }) 
+        
+        return render(request, self.template_name, {"form": user_edit_form, "profile_form": profile_edit_form})
+    
+    def post(self, request, *args, **kwargs):
+        
+        user_profile_data = self.get_user_profile_info(request)
+                        
+        user_form_details = self.user_edit_form_class(request.POST, instance=request.user, initial={
+                                                    "email" : user_profile_data.user.email,})
+        
+        profile_form_details = self.profile_edit_form_class(request.POST, instance=user_profile_data, initial={
+                                                   "affiliation" : user_profile_data.affiliation,})
+        
+        if user_form_details.is_valid() and profile_form_details.is_valid():
+                    
+            user_form_details.save()
+            profile_form_details.save()
+            
+            
+            #logger.info(user_form_details.cleaned_data, exc_info=True)
+            
+            logger.info("Updated First Name: "+str(self.get_user_profile_info(request).user.first_name), exc_info=True)
+            logger.info("Updated Last Name: "+str(self.get_user_profile_info(request).user.last_name), exc_info=True)
+            logger.info("Updated Department: "+str(self.get_user_profile_info(request).department), exc_info=True)
+ 
+            return render(request, "registration/edit_profile_done.html")
```

Comment:
I personally think, that redirect should lead back to the userprofile view.

Do you think that this page is necessary?

Reply:
Now redirect goes to profile view.